p2p/net/swarm/swarm_dial.go

```diff
@@ -382,6 +393,11 @@ func (s *Swarm) resolveAddrs(ctx context.Context, pi peer.AddrInfo) ([]ma.Multia
 			}
 		}
 
+		if len(toResolve) >= maximumToResolveAddresses {
+			// No point in resolving any more addresses, we don't have space.
+			continue
+		}
+
 		// otherwise, resolve it
 		reqaddr := addr.Encapsulate(p2paddr)
 		resaddrs, err := s.maResolver.Resolve(ctx, reqaddr)
```

Comment:
Do you know why we do this? Do we need the peer ID to resolve DNS addresses?

Reply:
I believe it's to assert that the dnsaddr we get has the matching peer id suffix (`/p2p/foo`). I don't love that it's so subtle